### ISSUE TITLE
Incorrect flake8-comprehension fixits (C401,C402) in fstring

### ISSUE THREAD

**Issue by Skylion007** [charliermarsh/ruff#2770]
One annoying false positive change in this flake8-comprehensions are set/dict conversions to literals in f'strings.

```python
b = list(range(10))
print(f'{set(a if a < 6 else 0  for a in b)}')
```
which prints: `{0, 1, 2, 3, 4, 5}`

becomes

```python
b = list(range(10))
print(f'{{a if a < 6 else 0  for a in b}}')
```
which is actually wrong if it's in a fstring since it actually just escapes the innermost curly brace. This results in the printing of `{a if a < 6 else 0 for a in b}`

It should fixit as 

```python
b = list(range(10))
print(f'{ {a if a < 6 else 0  for a in b} }')
```
The space will prevent the curly brace from being escaped and is deleted when the fstring resolves which should provide identical behavior to the original code snippit. See this Stackoverflow post: https://stackoverflow.com/questions/49120113/dictionary-set-comprehensions-inside-of-f-string

I suspect you wound find similar behavior for C402 with dict comprehensions.

Command, Version Info, and Settings
```
ruff --select C401,C402 example.py --fix
ruff 0.0.244
no ruff pyproject.toml ruff settings specified
```

**Comment by charliermarsh** [comment]
Ahh thank you! Good catch. I guess arguably we should avoid flagging this at all in f-strings.

**Comment by charliermarsh** [comment]
\cc @sbrugman if you're interested.

**Comment by Skylion007** [comment]
@charliermarsh It's not a bad change in fstrings, and it is potentially more performant, just need to be careful to do it properly :P . Another option is just to add an extra space around the inserted braces unconditionally and let black or whatever formatter you use deal with it.

**Comment by charliermarsh** [comment]
Yeah I could see either being "correct". Regardless sorry for the breakage!
